ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/signing/hashicorp/HashicorpSignerBuilder.java

```diff
@@ -0,0 +1,140 @@
+
+  public HashicorpSignerBuilder(final HashicorpSignerConfig config, final Vertx vertx) {
+    this.config = config;
+    this.vertx = vertx;
+    this.hashicorpVaultClient = vertx.createHttpClient();
+  }
+
+  public TransactionSigner build() {
+    final String tokenString = readTokenFromFile();
+    if (tokenString == null) {
+      return null;
+    }
+    final String response = requestSecretFromVault(tokenString);
+    if (response == null) {
+      return null;
+    }
+    final Credentials credentials = extractCredentialsFromJson(response);
+    if (credentials == null) {
+      return null;
+    }
+    return new CredentialTransactionSigner(credentials);
+  }
+
+  private String readTokenFromFile() {
+    final List<String> authFileLines;
+    try {
+      authFileLines = Files.readAllLines(config.getAuthFilePath());
+    } catch (IOException e) {
+      LOG.error(
+          "Unable to read file containing the authentication information for Hashicorp Vault.\n"
+              + config.getAuthFilePath()
+              + "\n"
+              + e.getMessage());
+      return null;
+    }
+    return authFileLines.get(0);
+  }
+
+  private String requestSecretFromVault(final String tokenString) {
+    final String requestURI = HASHICORP_SECRET_ENGINE_VERSION + config.getSigningKeyPath();
+
+    return getVaultResponse(tokenString, requestURI);
+  }
+
+  private String getVaultResponse(String tokenString, String requestURI) {
+    final CompletableFuture<String> future = new CompletableFuture<>();
+    final HttpClientRequest request =
+        hashicorpVaultClient.request(
+            HttpMethod.GET,
+            config.getServerPort(),
+            config.getServerHost(),
+            requestURI,
+            rh -> {
+              rh.bodyHandler(
+                  bh -> {
+                    if (rh.statusCode() == 200) {
+                      future.complete(bh.toString());
+                    } else {
+                      future.completeExceptionally(
+                          new Exception(
+                              "Hashicorp vault responded with status code {"
+                                  + rh.statusCode()
+                                  + "}"));
+                    }
+                  });
+            });
+    if (tokenString != null) {
+      request.headers().set("X-Vault-Token", tokenString);
+    }
+    request.setChunked(false);
+    request.end();
+    return getResponse(future);
+  }
+
+  private String getResponse(final CompletableFuture<String> future) {
+    String response;
+    try {
+      response = future.get(config.getTimeout(), TimeUnit.SECONDS);
+    } catch (InterruptedException | ExecutionException e) {
+      LOG.error("Unable to retrieve private key from Hashicorp Vault.", config.toString(), e);
+      return null;
+    } catch (TimeoutException e) {
+      LOG.error("Timeout while retrieving private key from Hashicorp Vault.", config.toString(), e);
```

Comment:
similar to above missing placeholder in log message for the config

Reply:
Same thing about not printing the exception.

Reply:
done